src/util/util.shared.ts

```diff
@@ -271,6 +271,23 @@ export const Utils = {
     }
   },
   delay: (t: number) => new Promise((resolve) => setTimeout(resolve, t)),
+  retryWithTimeout: async <T>(action: () => Promise<T>, timeoutInMs = 10000, delayInMs = 500): Promise<T> => {
```

Comment:
There is already timeout retry inside tatum.connector, why you create a new and not reuse or extend?